osu.Game/Rulesets/Edit/IPositionSnapProvider.cs

```diff
@@ -30,6 +30,8 @@ public interface IPositionSnapProvider
         /// </summary>
         /// <param name="referenceObject">An object to be used as a reference point for this operation.</param>
         /// <returns>The distance between two points residing in the timing point that are one beat length apart.</returns>
+        // todo: this, and other distance-related methods below it, should move to IDistanceSnapProvider.
```

Comment:
I can look into it as a follow-up to this PR, and if everyone here agrees on this split.

Reply:
Hmm, actually I'm not entirely sure how this can be properly work. The only idea I have for this is creating a separate `DistancedHitObjectComposer` which implements this interface, but does that seem like a good direction?

Reply:
Have pushed changes since that resolves both this conversation and the one about the existence of `OsuToolboxComposite` as well.